src/clients/HubPoolClient.ts

```diff
@@ -805,6 +805,17 @@ export class HubPoolClient extends BaseAbstractClient {
     return { runningBalance, incentiveBalance };
   }
 
+  public isValidChainId(_chainId: number, mainnetBlockToCheck: number): boolean {
+    // Get the earliest cross chain contract set event.
+    const earliestCrossChainContractEvent = this.crossChainContracts[_chainId]?.reduce((prev, curr) => {
```

Comment:
Why use `reduce` here? I was thinking we could use very similar logic to [this](https://github.com/across-protocol/sdk/blob/479f2992e4ae4301fff01da0d986b4d87229f8ec/src/clients/HubPoolClient.ts#L166)

Reply:
Yeah, good point. https://github.com/across-protocol/sdk/pull/842/commits/b57ac22cf3bff05267008a4460b0b981e46ff510